Source/WebKit/UIProcess/ProcessThrottler.cpp

```diff
@@ -211,7 +211,7 @@ void ProcessThrottler::updateThrottleStateIfNeeded()
         // If the process is currently runnable but will be suspended then first give it a chance to complete what it was doing
         // and clean up - move it to the background and send it a message to notify. Schedule a timeout so it can't stay running
         // in the background for too long.
-        if (m_state != ProcessThrottleState::Suspended) {
+        if (m_state != ProcessThrottleState::Suspended && !m_pendingRequestToSuspendID) {
```

Comment:
This change doesn't look right. We need to figure out what you changed recently to cause this crash and fix the underlying cause.

Your proposed fix would cause us to potentially release the process assertion while the PrepareToSuspend IPC handshake is still going on, AFAICT.

Reply:
The change that caused this is the newly added call to `updateThrottleStateIfNeeded()` in `setShouldTakeSuspendedAssertion()`. I think the right solution would actually be to remove the new call to `updateThrottleStateIfNeeded()` and instead just drop/take the suspended assertion if needed